src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_notification_update.html.twig

```diff
@@ -35,7 +35,7 @@
 {% block addon_description %}
   {% if module.attributes.changeLog is defined %}
     <ul>
-      {% for version, lines in module.attributes.changeLog %}
+      {% for version, lines in module.attributes.changeLog|arrayCast %}
```

Comment:
Suggestion / crazy idea: be able to handle potential future API changes by being able to parse `whatever the API sends to us`

How to do it ?
Parse the API data received into `module.attributes.changeLog` and have something like this:
```
if ($isJson) {
    return json_decode($data);
elseif ($isArray) {
    return arrayCast($data);
elseif ($isStdClass) {
    return objectCast($data);
}
```

This will make sure that, even if Addons API changes in 2 or 3 years, people using PS 1.7.8 will still be able to parse it.

The above ⏫ is a suggestion. Now I have a **mandatory** review feedback 😄 we must make sure that `even if Addons API is broken or changes we must now crash the page`
So can you test what happens if
- fetched data is `null`
- fetched data is bad JSON
- fetched data is JSON / XML / StdClass / array

Reply:
@matks I think everything is already handled by `AddonsDataProvider` which takes care of fetching data from addons but as you can see [here](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Service/DataProvider/Marketplace/ApiClient.php#L109), `json_decode` returns objects, not arrays.

Reply:
Gotcha 👍